src/queries/label-text.js

```diff
@@ -16,9 +16,17 @@ function queryAllLabelsByText(
 ) {
   const matcher = exact ? matches : fuzzyMatches
   const matchNormalizer = makeNormalizer({collapseWhitespace, trim, normalizer})
-  return Array.from(container.querySelectorAll('label')).filter(label =>
-    matcher(label.textContent, label, text, matchNormalizer),
-  )
+  return Array.from(container.querySelectorAll('label')).filter(label => {
+    let textToMatch = label.textContent
+
+    // The children of a textarea are part of `textContent` as well. We
+    // need to remove them from the string so we can match it afterwards.
+    label.querySelectorAll('textarea').forEach(textarea => {
+      textToMatch = textToMatch.replace(textarea.value, '')
```

Comment:
Hmm... What if the label is the same as the text value at that moment?

Reply:
I think we should be able to use `label.childNodes[0].textContent` here?

Reply:
@alexkrolick `replace` only replaces the first occurence:

```js
'abcabc'.replace('abc', '') // "abc"
```

Therefore this should be fine.

@timdeschryver Sorry, I can't follow? Generally I think it's good to avoid `childNodes` so we support arbitrary nesting for both the `textarea` and the label. There could be some divs and spans wrapped around for example.